src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprRecipeCategory.java

```diff
@@ -0,0 +1,133 @@
+import org.skriptlang.skript.bukkit.recipes.RegisterRecipeEvent;
+
+@Name("Recipe Category")
+@Description("The recipe category of a shaped, shapeless, blasting, furnace, campfire or smoking recipe.")
+@Examples({
+	"register a new shaped recipe with the key \"my_recipe\":",
+		"\tset the recipe ingredients to diamond, air, diamond, air, emerald, air, diamond, air and diamond",
+		"\tset the recipe category to crafting misc",
+		"\tset the recipe result item to nether star",
+	"",
+	"register a new blasting recipe with the id \"my_recipe\":",
+		"\tset the recipe input item to coal",
+		"\tset the recipe category to cooking misc",
+		"\tset the recipe result item to gunpowder",
+	"",
+	"loop the server's recipes:",
+		"\tbroadcast recipe category of loop-recipe"
+})
+@Since("INSERT VERSION")
+public class ExprRecipeCategory extends PropertyExpression<Recipe, RecipeCategory> {
+
+	static {
+		Skript.registerExpression(ExprRecipeCategory.class, RecipeCategory.class, ExpressionType.PROPERTY,
+			"[the] recipe category [of %recipes%]");
+	}
+
+	private boolean isEvent = false;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!exprs[0].isDefault()) {
+			//noinspection unchecked
+			setExpr((Expression<? extends Recipe>) exprs[0]);
+		} else {
+			if (exprs[0] == null) {
+				Skript.error("There is no recipe in a '" + getParser().getCurrentEventName() + "' event.");
+				return false;
+			}
+			if (getParser().isCurrentEvent(RegisterRecipeEvent.class))
+				isEvent = true;
+			setExpr(new EventValueExpression<>(Recipe.class));
+		}
+		return true;
+	}
+
+	@Override
+	protected RecipeCategory @Nullable [] get(Event event, Recipe[] source) {
+		return get(source, recipe -> {
+			Enum<?> category = null;
+			if (recipe instanceof MutableRecipe) {
+				if (recipe instanceof MutableCraftingRecipe craftingRecipeWrapper) {
+					category = craftingRecipeWrapper.getCategory();
+				} else if (recipe instanceof MutableCookingRecipe cookingRecipeWrapper) {
+					category = cookingRecipeWrapper.getCategory();
+				}
+			} else {
+				if (recipe instanceof ShapedRecipe shapedRecipe) {
+					category = shapedRecipe.getCategory();
+				} else if (recipe instanceof ShapelessRecipe shapelessRecipe) {
+					category = shapelessRecipe.getCategory();
+				} else if (recipe instanceof CookingRecipe<?> cookingRecipe) {
+					category = cookingRecipe.getCategory();
+				}
+			}
+			if (category != null)
+				return RecipeCategory.convertBukkitToSkript(category);
+			return null;
+		});
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET && isEvent) {
+			return CollectionUtils.array(RecipeCategory.class);
+		}
+		return null;
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		if (!(event instanceof RegisterRecipeEvent recipeEvent))
+			return;
```

Comment:
if change is limited to RRE, you should check that and error in acceptChange

Reply:
It is already there, this is just precautionary measure

Reply:
No, it should specify why it's erroring. Right now it just claims that you can't set it at all, which is false. It should specify that it can only be set in the event (and a precautionary measure for something that's specifically forbidden from happening doesn't need to exist)

Reply:
So, should I do a Skript.error within acceptChange? Dont know if theyre allowed there or if it's considered as a runtime error.

Reply:
See ExprHealAmount